### PR TITLE
chore(react): add typography variants

### PR DESCRIPTION
**Pull request by brionmario**
### Purpose

- Added new variants and modified the appearance of the existing Typography variants.

Screenshot 2023-02-22 at 01 39 52

### Related Issues
- N/A

### Related PRs
- N/A

### Checklist
- [ ] UX/UI review done on the final implementation.
- [ ] Story provided. (Add screenshots)
- [ ] Manual test round performed and verified.
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Documentation provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: packages/react/.storybook/preview.tsx [hunk 1]
```diff
@@ -64,7 +64,6 @@ addParameters({
     sort: 'requiredFirst',
   },
   docs: {
-    inlineStories: false,
```
Comment (savindi7) [inline packages/react/.storybook/preview.tsx:67]: why are we doing this?

File: packages/react/src/components/Typography/Typography.stories.mdx [hunk 2]
```diff
@@ -50,3 +52,37 @@ function Demo() {
...
+    {Object.entries(defaultTheme.typography)
+      .sort((a, b) => {
+        return b[1].fontSize?.split('rem')[0] - a[1].fontSize?.split('rem')[0];
+      })
+      .map(([key, value]) => {
+        const isVariant = key !== 'button' && (value.hasOwnProperty('fontSize') || value.hasOwnProperty('fontWeight'));
+        if (!isVariant) {
+          return null;
+        }
+        console.log(JSON.stringify(defaultTheme.typography));
```
Comment (thivi) [inline packages/react/src/components/Typography/Typography.stories.mdx:78]: Shall we remove this?

File: packages/react/src/models/theme.ts [hunk 3]
```diff
@@ -55,6 +55,33 @@ declare module '@mui/material/styles' {
...
     };
   }
+  interface TypographyVariants {
+    body3: CSSProperties;
+    body4: CSSProperties;
+    body5: CSSProperties;
+    display1: CSSProperties;
+    display2: CSSProperties;
+  }
+  interface TypographyVariantsOptions {
```
Comment (thivi) [inline packages/react/src/models/theme.ts:65]:
```suggestion

  interface TypographyVariantsOptions {
```

File: packages/react/.storybook/preview.tsx (hunk 1 above)
Comment (brionmario) [reply packages/react/.storybook/preview.tsx:67]: This was added to overcome the stories leaking outside the iframe in components with `fixed` styling.

Screenshot 2023-02-22 at 12 41 38

However, this causes the default maxheight of the story window to be removed.

Screenshot 2023-02-22 at 12 40 33

Hence, decided to revert the config globally and add in stories selectively.

File: packages/react/src/models/theme.ts (hunk 3 above)
Comment (brionmario) [reply packages/react/src/models/theme.ts:66]: Fixed with https://github.com/wso2/oxygen-ui/pull/40/commits/1ac617fbe10c03596f88e8c98c2dc105d7cea938

File: packages/react/src/components/Typography/Typography.stories.mdx (hunk 2 above)
Comment (brionmario) [reply packages/react/src/components/Typography/Typography.stories.mdx]: Fixed with https://github.com/wso2/oxygen-ui/pull/40/commits/1ac617fbe10c03596f88e8c98c2dc105d7cea938

**Review by savindi7 — APPROVED** [review]

**Pull request merged**